topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.md

```diff
@@ -0,0 +1,471 @@
+
+![Dividing nuclei](../../images/object-tracking-using-cell-profiler/Dividing_nuclei.gif "Time lapse recording of anaphase nuclei in a Drosophila embryo.")
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> 1. Create a new history for this tutorial.  
+>    When you log in for the first time, an empty, unnamed history is created by default. You can simply rename it.
+>    {% snippet snippets/create_new_history.md %}
+>
+> 2. Import {% icon galaxy-upload %} the files from [Zenodo]({{ page.zenodo_link }}) or from
+>    the shared data library.
+>    - **Important:** If setting the type to 'Auto-detect', make sure that after upload, the datatype is set to zip.
+>
+>    ```
+>    https://zenodo.org/api/files/e5d1bd5c-60a0-42e4-8f0d-a2ebc863c5d9/drosophila_sample.zip
+>    ```
+>
+>    {% snippet snippets/import_via_link.md %}
+>
+> 3. Rename {% icon galaxy-pencil %} the file to drosophila_embryo.zip
+>
+{: .hands_on}
+
+
+# Create a CellProfiler pipeline in Galaxy
+
+In this section, we will build a CellProfiler pipeline from scratch in Galaxy.
+We need to:  
+  - Read the images and the metadata  
+  - Convert the colour images to grayscale  
+  - Segment the nuclei  
+  - Extract features from the segmented nuclei  
+  - Perform tracking  
+  - Produce some useful output files  
+
+A pipeline is built by chaining together Galaxy tools representing CellProfiler modules and must start with the {% tool [Starting modules](toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1) %} tool and end with the {% tool [CellProfiler](toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0) %}  tool.  
+
+
+
+![Image of the workflow](../../images/object-tracking-using-cell-profiler/CP_object_tracking_pipeline.png "Overview of the CellProfiler pipeline using Galaxy tools.")
+
+
+> ### {% icon details %} More details about the pipeline steps
+>    - Metadata is needed to tell CellProfiler what a temporal sequence of images is and what the order of images is in the sequence.
+>    - CellProfiler is designed to work primarily with grayscale images. Since we don't need the colour information, we convert colour images to grayscale type.  
+>    - Segmentation means identifying the nuclei in each image. In CellProfiler, this is done by thresholding the intensity level in each image.  
+>    - When we perform tracking we're usually interested in quantifying how some properties of the objects evolve over time. Also, sometimes we may want to do tracking by matching objects based on some property of the objects (e.g. a shape measurement). Therefore, for each segmented object, we compute some features, i.e. numerical descriptors of some properties of the object. 
+>    - Tracking will provide the information required to allow downstream data analysis tools to link the features into a multidimensional time series. 
+> 
+>
+{: .details}
+
+## Create a new workflow
+> 1. Create a new workflow
```

Comment:
so I see a step to create a WF, but I don't see that it's saved or run at any step?

Is this intended?

Reply:
fixed in @jkh1's commit :)

Reply:
so, for me that doesn't really fix it? Why are they done in the WF editor? Is it to run a WF at the end? When do you exit the WF editor? When do you run the WF? This is the part that's not clear.

Reply:
yes, so we need to run the complete workflow at the end, individual tools can't be run here... We'll think about how to make it clearer in the text

Reply:
Discussed out of band: @beatrizserrano explained to me that CellProfiler is extremely strange and un-galaxy like. It makes more sense now.

Reply:
https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html#data-cleaning has a box for running a WF. Maybe you could do that?

Reply:
nice! that's what I was after, will add it :)

Reply:
yes, I included it yesterday as a requirement in the header. Also, a box explaining why that's important.